torchvision/models/maxvit.py

```diff
@@ -0,0 +1,854 @@
+                    activation_fn=activation_fn,
+                    head_dim=head_dim,
+                    mlp_ratio=mlp_ratio,
+                    mlp_dropout=mlp_dropout,
+                    attn_dropout=attn_dropout,
+                    partition_size=partition_size,
+                    grid_size=self.grid_size,
+                    p_stochastic_dropout=p,
+                ),
+            ]
+
+    def forward(self, x: Tensor) -> Tensor:
+        """
+        Args:
+            x (Tensor): Input tensor of shape (B, C, H, W).
+        Returns:
+            Tensor: Output tensor of shape (B, C, H, W).
+        """
+        for layer in self.layers:
+            x = layer(x)
+        return x
+
+
+class MaxVit(nn.Module):
+    """
+    Implements MaxVit Transformer from the `MaxViT: Multi-Axis Vision Transformer <https://arxiv.org/abs/2204.01697>`_ paper.
+    Args:
+        input_size (Tuple[int, int]): Size of the input image.
+        input_channels (int): Number of input channels.
+        stem_channels (int): Number of channels in the stem.
+        num_classes (int): Number of classes.
+        block_channels (List[int]): Number of channels in each block.
+        block_layers (List[int]): Number of layers in each block.
+        stochastic_depth_prob (float): Probability of stochastic depth. Expands to a list of probabilities for each layer that scales linearly to the specified value.
+        squeeze_ratio (float): Squeeze ratio in the SE Layer.
+        expansion_ratio (float): Expansion ratio in the MBConv bottleneck.
+        normalization_fn (Callable[..., nn.Module]): Normalization function.
+        activation_fn (Callable[..., nn.Module]): Activation function.
+        head_dim (int): Dimension of the attention heads.
+        mlp_ratio (int): Expansion ratio of the MLP layer.
+        mlp_dropout (float): Dropout probability for the MLP layer.
+        attn_dropout (float): Dropout probability for the attention layer.
+        partition_size (int): Size of the partitions.
+    """
+
+    def __init__(
+        self,
+        # input size parameters
+        input_size: Tuple[int, int],
+        # stem and task parameters
+        input_channels: int,
+        stem_channels: int,
+        num_classes: int,
+        # block parameters
+        block_channels: List[int],
+        block_layers: List[int],
+        stochastic_depth_prob: float,
+        # conv parameters
+        squeeze_ratio: float,
+        expansion_ratio: float,
+        # conv + transformer parameters
+        # normalization_fn is applied only to the conv layers
+        # activation_fn is applied both to conv and transformer layers
+        normalization_fn: Callable[..., nn.Module],
+        activation_fn: Callable[..., nn.Module],
+        # transformer parameters
+        head_dim: int,
+        mlp_ratio: int,
+        mlp_dropout: float,
+        attn_dropout: float,
+        # partitioning parameters
+        partition_size: int,
```

Comment:
Quite a few parameters with several non-default configs. Consider adding some good default values for some of these configurations.

Also usually things like `squeeze_ratio`, `expansion_ratio` are constant across model variants and usually get hardcoded. These parameters are often configurable in research repos as a means to reproduce the ablation experiments. When we clean up the implementations, typically we reduce the number of parameters to those necessary to differentiate across canonical variants.

Perhaps it's worth looking into Swin Transformer or ViT to see which parameters are usually set to defaults and the naming/order convention across the parameters. For example `num_classes=1000` or `normalization_fn` is usually called `norm_layer`.